src/components/astra-editor/plugins/prompt.ts

```diff
@@ -45,56 +47,45 @@ class PromptWidget extends WidgetType {
     // when we want to generate another suggestion or reject suggestion
     let previousSuggestion = ''
 
-    const container = document.createElement('div')
-    container.className = 'prompt-container'
-
-    const buttonGenerateContent = 'Generate <i>⌘ + ↩</i>'
-    container.innerHTML = `
-      <div class='prompt-wrap'>
-        <textarea class='prompt-fake-editor' style='visibility:hidden; position:absolute; height:1px;'></textarea>
-        <textarea class='prompt-editor' style='height:50px;' placeholder='I am EZQL, your personal database assistant, how can I help you?'></textarea>
-        <div class='prompt-error' style='display:none'></div>
-        <div class='prompt-action'>
-          <div>
-            <button id='prompt-generate'>${buttonGenerateContent}</button>
-          </div>
-          <div>
-            <button id='prompt-reject'>Reject <i>⌘ + Z</i></button>
-          </div>
-          <div style='flex-grow:1; justify-content: end; align-items:center;'>Esc to close</div>
-        </div>
-      </div>
-    `
-
-    const input = container.querySelector('.prompt-editor') as HTMLTextAreaElement
+    const container = document.createElement('astra-editor-prompt-dialog') as PromptDialog
 
-    const fakeInput = container.querySelector('.prompt-fake-editor') as HTMLTextAreaElement
+    container.theme = this.plugin.editor.color ?? 'light'
 
-    const errorMessage = container.querySelector('.prompt-error') as HTMLDivElement
+    this.themeHandler = () => {
+      container.theme = this.plugin.editor.color ?? 'light'
+    }
 
-    const generateButton = container.querySelector('#prompt-generate') as HTMLButtonElement
+    this.plugin.editor.addEventListener('color-changed', this.themeHandler)
 
-    const rejectButton = container.querySelector('#prompt-reject') as HTMLButtonElement
+    container.addEventListener('close', () => {
+      view.dispatch({ effects: [effectHidePrompt.of(null)] })
+      setTimeout(() => view.focus(), 50)
+    })
```

Comment:
do we need to worry about removing these / multiple getting attached? I was expecting to maybe see more `container.removeEventListener`s somewhere

Reply:
https://stackoverflow.com/questions/12528049/if-a-dom-element-is-removed-are-its-listeners-also-removed-from-memory

Event handler will be removed if the DOM is removed and no reference pointing to it.